lib/matplotlib/collections.py

```diff
@@ -1001,7 +1001,7 @@ def set_sizes(self, sizes, dpi=72.0):
 
     @artist.allow_rasterization
     def draw(self, renderer):
-        self.set_sizes(self._sizes, self.figure.dpi)
+        self.set_sizes(self._sizes, self.get_figure(root=False).dpi)
```

Comment:
Subfigures ask their parent for the dpi anyway so just go straight there and save a property call.

```suggestion
        self.set_sizes(self._sizes, self.get_figure(root=True).dpi)
```